### ISSUE TITLE
subprocess modernize rules

### ISSUE THREAD

**Issue by kaddkaka** [astral-sh/ruff#19623]
### Summary

subprocess package has a "new" and "old" API, see the in-source docstring below.

I think it would be valuable to have lint rules that warn when using the old API and recommend using the new one instead.

```py
Main API
========
run(...): Runs a command, waits for it to complete, then returns a
          CompletedProcess instance.
Popen(...): A class for flexibly executing a command in a new process

Constants
---------
DEVNULL: Special value that indicates that os.devnull should be used
PIPE:    Special value that indicates a pipe should be created
STDOUT:  Special value that indicates that stderr should go to stdout


Older API
=========
call(...): Runs a command, waits for it to complete, then returns
    the return code.
check_call(...): Same as call() but raises CalledProcessError()
    if return code is not 0
check_output(...): Same as check_call() but returns the contents of
    stdout instead of a return code
getoutput(...): Runs a command in the shell, waits for it to complete,
    then returns the output
getstatusoutput(...): Runs a command in the shell, waits for it to complete,
    then returns a (exitcode, output) tuple
```

**Comment by MichaReiser** [comment]
Reference: https://docs.python.org/3/library/subprocess.html#older-high-level-api
